JSONRenderer.php

```diff
@@ -44,6 +47,14 @@ public function renderReport(Report $report)
                 );
             }
 
+            $name = $violation->getMethodName() ?: $violation->getFunctionName() ?: $violation->getClassName();
```

Comment:
😨 lots of ternary stacking. We avoid this operator in other languages: I realize we don't really have a real style for PHP, but should we also avoid it here?